core/zip/inc/Compression.h

```diff
@@ -55,14 +58,37 @@ enum ECompressionAlgorithm {
    kUndefinedCompressionAlgorithm
 };
 
-/// Deprecated name, do *not* use:
-static const enum ECompressionAlgorithm kUseGlobalSetting = kUseGlobalCompressionSetting;
+enum ECompressionLevel {
```

Comment:
Hi @Axel-Naumann -

Indeed, this is the goal.  Here's how we approached it in the prototype branch:

https://github.com/bbockelm/root/blob/use_global_compression_defaults_example/core/zip/inc/Compression.h#L83

The class-based approach allows a gradual transition to type safety (further, it can be done without breaking ABI or API!).  By allowing implicit conversion -- but marking these as deprecated -- users will get a warning when they use an integer instead of specifying a proper enum member.

Brian